src/mds/StrayManager.cc

```diff
@@ -573,6 +573,17 @@ bool StrayManager::eval_stray(CDentry *dn, bool delay)
           (*p)->try_remove_dentries_for_stray();
         }
       }
+
+      if (!in->remote_parents.empty()) {
+	// unlink any stale remote snap dentry.
+	for (compact_set<CDentry*>::iterator p = in->remote_parents.begin();
+	     p != in->remote_parents.end(); ) {
+	  CDentry *remote_dn = *p;
+	  ++p;
+	  assert(remote_dn->last != CEPH_NOSNAP);
+	  remote_dn->unlink_remote(remote_dn->get_linkage());
```

Comment:
There are a bunch of prior checks in this function, but I'm not sure which of them are "making sure the stray directory inode is not referenced by any snapshot". Can you point me at it?
The logic in some of these snapshot functions is very obfuscated! :(